app/src/main/java/org/simple/clinic/bp/assignbppassport/BpPassportSheet.kt

```diff
@@ -77,9 +83,27 @@ class BpPassportSheet : BottomSheetActivity(), BpPassportUiActions {
   }
 
   private val bpPassportIdentifier: Identifier by lazy {
-    intent.getParcelableExtra(KEY_BP_PASSPORT_NUMBER)!!
+    screenKey.identifier
+  }
+
+  override fun defaultModel() = BpPassportModel.create(bpPassportIdentifier)
+
+  override fun bindView(inflater: LayoutInflater, container: ViewGroup?): SheetBpPassportBinding {
+    return SheetBpPassportBinding.inflate(layoutInflater)
+  }
+
+  override fun events(): Observable<BpPassportEvent> {
+    return Observable
+        .merge(
+            registerNewPatientClicks(),
+            addToExistingPatientClicks()
+        )
```

Comment:
We would need to report these to analytics. I see that this was not done in the original implementation as well, so analytics events from this screen are absent even in the latest build.